views/urls.go

```diff
@@ -3,15 +3,63 @@ package views
 import (
 	"net/url"
 
+	"github.com/a-h/templ"
 	"github.com/ugent-library/bind"
 )
 
-func urlWithQuery(u *url.URL, v any) *url.URL {
-	vals, err := bind.EncodeQuery(v)
+type URLBuilder struct {
+	url *url.URL
+}
+
+func NewURLBuilder(base string) *URLBuilder {
```

Comment:
we're going to use this a lot so maybe add a less verbose `function URL() *URLBuilder` constructor alias? And maybe also `Path` instead of `WithPath` etc?

Reply:
Done. Also renamed WithQuery to Query and AddQuery to AddQueryParam.